### PR TITLE
subscription-manager pre-registration

### PR DESCRIPTION
**Pull request by brianwcook**
This add support for pre-registering with subscription-manager in the buildah tasks (0.2 only). When activation keys are provided the buildah-task container will register and then provide certs to the buildah env via volume mounts so that users do not need to add subscription-manager register commands to the Containerfiles. This makes downstreaming somewhat easier and addresses negative user feedback.

# Before you complete this pull request ...

Look for any open pull requests in the repository with the title "e2e-tests update" and 
see if there are recent e2e-tests updates that will be applicable to your change.

### CODE REVIEW

**Comment by skateman** [comment]
Tested with the buildah task, works well 🎉

**Comment by brianwcook** [comment]
opened https://github.com/konflux-ci/buildah-container/pull/98 to get subscription-manager included in buildah-task

**Comment by ezr-ondrej** [comment]
Works on ubi9 as well :)

**Comment by brianwcook** [comment]
rebased and removed dnf install.

File: task/buildah-oci-ta/0.2/buildah-oci-ta.yaml
Comment (chmeliik) [inline task/buildah-oci-ta/0.2/buildah-oci-ta.yaml:438]: Without this check, would something we mount break the ability to `subcription-manager register`?

The string check is very flimsy, I would rather drop it if possible

(In fact, currently the check doesn't work at all, it's always `grep: register: No such file or directory`)

File: task/buildah-oci-ta/0.2/buildah-oci-ta.yaml
Comment (chmeliik) [inline task/buildah-oci-ta/0.2/buildah-oci-ta.yaml:451]:
```suggestion
            trap 'subscription-manager unregister' EXIT
```
This will make us unregister always, even if the build fails

File: task/buildah-oci-ta/0.2/buildah-oci-ta.yaml
Comment (chmeliik) [inline task/buildah-oci-ta/0.2/buildah-oci-ta.yaml:457]: So if I understand correctly:
* we are generating entitlement certificates in a Fedora container image (the one that runs this script)
* the Fedora image installs `subscription-manager`, which comes with the `redhat-uep.pem` file
* we're mounting those into the build, which could be based on ubi8, ubi9, ubiX-minimal, ...

@ptoscano could you check if we're doing something sensible here?

**Comment by brianwcook** [comment]
i'm doing some add'l testing, don't merge for now.

**Comment by brianwcook** [comment]
this is ready to go. tested with:
* entitlement certificates as key value secrets
* activation keys secret + registration in Dockerfile
* activation keys as secrets and auto-pre-registration

**Review by arewm — APPROVED** [review]
LGTM

File: task/buildah-oci-ta/0.2/buildah-oci-ta.yaml
Comment (brianwcook) [reply task/buildah-oci-ta/0.2/buildah-oci-ta.yaml:438]: I made it more robust and fixed the bug before merging.

File: task/buildah-oci-ta/0.2/buildah-oci-ta.yaml
Comment (brianwcook) [reply task/buildah-oci-ta/0.2/buildah-oci-ta.yaml:451]: did not get this in but I will file a follow up PR. testing all cases manually was getting complicated

File: task/buildah-oci-ta/0.2/buildah-oci-ta.yaml
Comment (brianwcook) [reply task/buildah-oci-ta/0.2/buildah-oci-ta.yaml:457]: I _think_ that the redhat-uep.pem is actually generated during the registration, but the rest of what you say is accurate

Comment (ptoscano) [reply task/buildah-oci-ta/0.2/buildah-oci-ta.yaml:457]: > I _think_ that the redhat-uep.pem is actually generated during the registration,

```bash
$ rpm -ql subscription-manager-rhsm-certificates 
/etc/rhsm
/etc/rhsm/ca
/etc/rhsm/ca/redhat-entitlement-authority.pem
/etc/rhsm/ca/redhat-uep.pem
/usr/share/licenses/subscription-manager-rhsm-certificates
/usr/share/licenses/subscription-manager-rhsm-certificates/COPYING
```

Comment (ptoscano) [reply task/buildah-oci-ta/0.2/buildah-oci-ta.yaml:457]: @chmeliik other than what I wrote above about "where redhat-uep.pem comes from", what you said seems correct

Comment (chmeliik) [reply task/buildah-oci-ta/0.2/buildah-oci-ta.yaml:457]: And can it cause some problems? Mounting `redhat-uep.pem` from the Fedora version of the package? Or does it generally not matter?

File: task/buildah-oci-ta/0.2/buildah-oci-ta.yaml
Comment (ptoscano) [inline task/buildah-oci-ta/0.2/buildah-oci-ta.yaml:449]: this ought to exclude comments, otherwise a line such as
```
# install packages with no need to register using subscription-manager
dnf -y install hello
```
will trigger this detection to think the dockerfile is registering on its own

File: task/buildah-oci-ta/0.2/buildah-oci-ta.yaml
Comment (ptoscano) [reply task/buildah-oci-ta/0.2/buildah-oci-ta.yaml:457]: It should not be an issue, since `redhat-uep.pem` is shipped by `subscription-manager-rhsm-certificates`, which would need to be upgraded in case any of the certificates it ships needs to be changed. In that case, the update of the package ought to be done enough time in advance to ensure all the users (RHEL and not only, Fedora included) get the new package installed.

**Pull request merged**
